### PR TITLE
2025 migration

### PR DESCRIPTION
**Pull request by dgorbatov**

### CODE REVIEW

File: Controllers/OI/Joysticks.java [hunk 1]
```diff
@@ -20,7 +20,7 @@ public static Joysticks getInstance() {
 
     public static final class OIConstants  {
         public static final int DRIVE_PORT = 0;
-        public static final int TRANS_PORT = 0;
+        public static final int TRANS_PORT = 1;
```
Comment (Aditya9246) [inline Controllers/OI/Joysticks.java:23]: considering that we are moving to one xbox controller for all robot movement & one for operator controls, do we need a separate translation port?

File: Motor/MotorHelper.java [hunk 2]
```diff
@@ -2,20 +2,26 @@
...
 import edu.wpi.first.math.controller.PIDController;
 
 /**
  * Create Motor Functions deprecated as of Novemeber 2023, use MotorConfig instead
  */
 public class MotorHelper {
-    public static void setConversionFactor(SparkMax motor, double factor) {
-        motor.getEncoder().setPositionConversionFactor(factor);
-        motor.getEncoder().setVelocityConversionFactor(factor / 60);
+    @Deprecated
```
Comment (Aditya9246) [inline Motor/MotorHelper.java:16]: do we need to keep the older deprecated versions of these functions? it's generally good practice to remove them, unless there is a use for them

File: Controllers/OI/Joysticks.java (hunk 1 above)
Comment (dgorbatov) [reply Controllers/OI/Joysticks.java:23]: thats for the joysticks class, assuming we ever want to use joysticks it allows us to switch easily using the OI base class

File: Motor/MotorHelper.java (hunk 2 above)
Comment (dgorbatov) [reply Motor/MotorHelper.java:16]: previous code repos, ex: 2024 offseason robot code might not work would keep it around for a year and then remove it

Comment (Aditya9246) [reply Motor/MotorHelper.java:16]: alrighty

File: Swerve/SwerveConstants/SwerveModuleConstants.java [hunk 3]
```diff
@@ -133,46 +138,34 @@ public static SwerveModuleConstants[] generateConstants(Rotation2d[] angleOffset
...
+        return driveConfig.setConfig(new SparkMax(driveMotorId, MotorType.kBrushless));
     }
 
     public TalonFX getDriveFalcon() {
-        TalonFX drive = new TalonFX(driveMotorId, Constants.Ports.CANIVORE_NAME);
-        TalonFXConfiguration config = driveConfig.getTalonConfig();
-        config.Feedback.SensorToMechanismRatio = moduleInfo.driveGearRatio;
-        drive.getConfigurator().apply(config);
-        return drive;
+        return driveConfig.setConfig(new TalonFX(driveMotorId, Constants.Ports.CANIVORE_NAME));
```
Comment (Powerlax) [inline Swerve/SwerveConstants/SwerveModuleConstants.java]: Dont use Constants from frc.robot, they might not use the same conventions all the time (aka i pulled and vscode gave error).

Comment (Powerlax) [reply Swerve/SwerveConstants/SwerveModuleConstants.java]: Also all Constants.Ports are like this

**Review by Powerlax — CHANGES_REQUESTED** [review]
look at comments.

also your package statements are wrong (you say POPLib when it should be packofpartslib)

**Review by Powerlax — APPROVED** [review]
good code.

definety did not work on this

**Comment by Powerlax** [comment]
@dgorbatov  so apprently there was a slight goof and the code in here got merged in with the vision code

basically the code from this branch is already in main

should we delete the pr?

**Pull request closed without merging**